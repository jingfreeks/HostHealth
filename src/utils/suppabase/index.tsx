export {default as supabase} from './suppabase';
