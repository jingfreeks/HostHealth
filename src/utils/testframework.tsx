import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
// import {QueryClient, QueryClientProvider} from '@tanstack/react-query';
// // import {DARK_THEME} from '@/utils/constants';
// // import {ThemeProvider} from 'styled-components/native';

type MockProviderType = {
  children?: React.ReactNode;
  store?: object;
};
export const MockProvider = (props: MockProviderType) => {
  const {store, children} = props;
  const mockStore = configureStore([thunk]);
  const mocked = mockStore(store);
  return <Provider store={mocked}>{children}</Provider>;
};

export const testingProps = (reference: string): unknown => ({
  testID: reference,
  accessibilityLabel: reference,
});

// const queryClient = new QueryClient({
//   defaultOptions: {
//     queries: {
//       retry: false,
//     },
//   },
// });
// export const wrapper = ({children}) => (
//   <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
// );
