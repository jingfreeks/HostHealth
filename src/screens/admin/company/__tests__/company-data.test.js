import React from 'react';
import {fireEvent} from '@testing-library/react-native';
import {renderWithProviders} from '@/utils/testframeworknew';
import {Company} from '../index';

const deleteCompany = () => ({data: {}});
jest.mock('@/slice', () => {
  return {
    useGetCompanyQuery: () => ({
      isLoading: false,
      isSuccess: true,
      isError: false,
      company: {
        __v: 0,
        _id: '655f656dbed03fb0a5baddf0',
        name: 'San Francisco',
        address: 'test address',
        state: '655f48446bdacc1843348d7a',
      },
      data: {
        entities: {
          '655f656dbed03fb0a5baddf0': {
            __v: 0,
            _id: '655f656dbed03fb0a5baddf0',
            name: 'San Francisco',
            address: 'test address',
            state: '655f48446bdacc1843348d7a',
          },
        },
        ids: ['655f656dbed03fb0a5baddf0'],
      },
      selectFromResult:jest.fn()
    }),
    useDeleteCompanyMutation: () => [
      deleteCompany,
      {isLoading: false, isError: false},
    ],
  };
});
describe('Company admin with data screen', () => {
  it('Should work as expected to get snapshot', () => {
    const all = renderWithProviders(<Company />);
    expect(all.toJSON()).toMatchSnapshot();
  });

  it('Should to trigger Create Button', () => {
    const all = renderWithProviders(<Company />);
    const el = all.getByTestId('CompanyCreateButtonTestId');
    fireEvent(el, 'onPress');
    expect(all.toJSON()).toBeTruthy();
  });

  it('Should to trigger Edit Button', () => {
    const all = renderWithProviders(<Company />);
    const el = all.getByTestId('CompanyEditFormTestId');
    fireEvent(el, 'onPress');
    expect(all.toJSON()).toBeTruthy();
  });

  it('Should to trigger Delete Item Button', () => {
    const all = renderWithProviders(<Company />);
    const el = all.getByTestId('CompanyDeleteFormTestId');
    fireEvent(el, 'onPress');
    expect(all.toJSON()).toBeTruthy();
  });
});
