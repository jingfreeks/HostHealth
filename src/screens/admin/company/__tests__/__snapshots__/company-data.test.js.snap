// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company admin with data screen Should work as expected to get snapshot 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginBottom": 10,
      },
    ]
  }
>
  <RCTScrollView
    data={
      [
        "655f656dbed03fb0a5baddf0",
      ]
    }
    extraData={
      [
        "655f656dbed03fb0a5baddf0",
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "paddingBottom": 10,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "height": 196.17647058823528,
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                  },
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  [
                    {
                      "color": "#0e3f6c",
                      "fontFamily": "Avenir",
                      "fontSize": 39.23529411764706,
                      "fontWeight": "900",
                    },
                  ]
                }
              >
                San Francisco
              </Text>
              <Text
                numberOfLines={1}
                style={
                  [
                    {
                      "color": "#0e3f6c",
                      "fontFamily": "Avenir",
                      "fontSize": 16,
                      "fontWeight": "400",
                    },
                  ]
                }
              >
                test address
              </Text>
              <Text
                numberOfLines={1}
                style={
                  [
                    {
                      "color": "#0e3f6c",
                      "fontFamily": "Avenir",
                      "fontSize": 39.23529411764706,
                      "fontWeight": "900",
                    },
                  ]
                }
              >
                655f48446bdacc1843348d7a
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <View
                accessibilityLabel="CompanyEditFormTestId"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderRadius": 0,
                    "opacity": 1,
                    "paddingBottom": 15,
                    "paddingLeft": 15,
                    "paddingRight": 15,
                    "paddingTop": 15,
                  }
                }
                testID="CompanyEditFormTestId"
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": undefined,
                        "fontSize": 25,
                      },
                      undefined,
                      {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                accessibilityLabel="CompanyDeleteFormTestId"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderRadius": 0,
                    "opacity": 1,
                    "paddingBottom": 15,
                    "paddingLeft": 15,
                    "paddingRight": 15,
                    "paddingTop": 15,
                  }
                }
                testID="CompanyDeleteFormTestId"
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": undefined,
                        "fontSize": 25,
                      },
                      undefined,
                      {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "borderRadius": 28,
        "bottom": 0,
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 16,
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "transform": [
          {
            "scale": 1,
          },
        ],
      }
    }
  >
    <View
      style={
        [
          {
            "borderRadius": 3,
            "overflow": "hidden",
          },
          {
            "borderRadius": 3,
          },
          {
            "borderRadius": 28,
            "elevation": 4,
          },
          false,
        ]
      }
    >
      <View
        accessibilityLabel="CompanyCreateButtonTestId"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": false,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="CompanyCreateButtonTestId"
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ca71eb",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "center",
              "padding": 8,
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#ffffff",
                "fontSize": 18,
                "marginHorizontal": 8,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Create
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
